moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js

```diff
@@ -0,0 +1,153 @@
+  let featureCollection = {
+    type: "FeatureCollection",
+    features: [],
+  };
+  if (signal) {
+    const featureUUID = uuidv4();
+    const feature = {
+      type: "Feature",
+      properties: {
+        ...signal.properties,
+        renderType: "Point",
+        PROJECT_EXTENT_ID: featureUUID,
+        sourceLayer: "drawnByUser",
+      },
+      geometry: signal.geometry,
+      id: featureUUID,
+    };
+    featureCollection.features.push(feature);
+  }
+  return featureCollection;
+};
+
+/**
+ * Material Autocomplete wrapper that enables selecting a traffic/phb signal record from a 
+ * Socrata dataset. Data is fetched once when the component mounts.
+ * @param {Object} signal - A GeoJSON feature or a falsey object (e.g. "" from empty input)
+ * * @param {func} setSignal - signal state setter
+ * * @param {Object} projectDetails - The parent view's project details object
+ * * @param {Object} setProjectDetails - The projectDetails state setter
+ * * @param {Object} setFeatureCollection - The parent view's featureCollection state setter
+ * * @param {Boolean} signalError - If the current signal value is in validation error
+ *  @return {JSX.Element}
+ */
+const SignalAutocomplete = ({
+  signal,
+  setSignal,
+  projectDetails,
+  setProjectDetails,
+  setFeatureCollection,
+  signalError,
+}) => {
+  const [data, setData] = React.useState(null);
+  const [error, setError] = React.useState(null);
+
+  const handleFieldChange = signal => {
+    const projectName = signal?.properties?.location_name || "";
+    const updatedProjectDetails = {
+      ...projectDetails,
+      ...{ project_name: projectName },
+    };
+    setProjectDetails(updatedProjectDetails);
+    setSignal(signal);
+    const featureCollection = signalToFeatureCollection(signal);
+    setFeatureCollection(featureCollection);
+  };
+  const loading = !data && !error;
+
+  React.useEffect(() => {
+    const url =
+      "https://data.austintexas.gov/resource/p53x-x73x.geojson?$select=signal_id,location_name,location,signal_type&$order=signal_id asc&$limit=9999";
+    fetch(url)
+      .then(response => response.json())
+      .then(
+        result => {
+          if (result.error) {
+            // on query error, socrata returns status 200 with {"error": true, "message": <message>} in body
+            setError(result.message.toString());
+          } else {
+            // insert an empty option for the initialized (empty) state this creates a weird
+            // blank list option in the autocomplete menu. i have not found an alternative
+            // that avoids material linter errors
+            result.features.unshift("");
+            setData(result.features);
+          }
+        },
+        error => {
+          setError(error.toString());
+        }
+      );
+  }, []);
+
+  return (
+    <Autocomplete
+      id="signal-id"
+      filterOptions={(options, { inputValue, getOptionLabel }) => {
+        // limits options to ensure fast rendering
```

Comment:
I looked to see if it was possible, but we could do a virtualized list in the autocomplete if we wanted to. I can't imagine someone scrolling forever to find their signal though.

Reply:
oooh cool idea—I will read up on that